### DISCUSSION TITLE
whitebox_tools requires GLIBC 2.32 or higher

### DISCUSSION THREAD

**Discussion by ColoradoJake1** [opengeos/whitebox-python#55] [General]
Hi folks,

We have a docker container and we simply have whitebox via pip and it has worked for months. Starting somewhere between Wednesday night and yesterday, all our of servers and containers are now erroring out on whitebox. We are just using the simple python whitebox package, which appears to call out to get whitebox_tools on demand.

The errors:
(short form):    ./whitebox_tools: /lib/x86_64-linux-gnu/libc.so.6: version 'GLIBC_2.35' not found (required by ./whitebox_tools)
We get four of these lines (one for GLIBC 2.32 to 2.35).   

We have played with set_whitebox_dir.  We have a couple of us looking into this for two days and am out of ideas.

We do want to upgrade GLIBC (too many side effects). We have checked permissions, updates Ubuntu updates that might have triggered this problem starting last week. We tried dropping from whitebox 2.2.0, 2.0.0 and the 2.3.0 (that was released today).  We do not pull down and pre-compile the WBT engine??? but maybe we need to do it.  

Any guesses on why it is suddenly stopped?  

And how do we fix it?

Rob

**Comment by giswqs** [comment]
It is because `pip install whitebox` will always pull the latest WBT backend from the whitebox.com server at this url https://www.whiteboxgeo.com/WBT_Linux/WhiteboxTools_linux_amd64.zip. It does not matter what pip whitebox version you install, the program will always pull the latest WBT backend from this same url. I am not sure if @jblindsay keeps a copy of the old version WBT on the whitebox.com server.

One temporary solution is that you pull the WBT backend from the URL below. It is the one I built using Ubuntu 18.04 with an older version of GLIBC so that users can use it with Google Colab. In your docker, you download this zip file and unzip it to your whitebox package installation directory, e.g., `python3.x/site-packages/whitebox/whitebox_tools`
https://github.com/giswqs/whitebox-bin/blob/master/WhiteboxTools_ubuntu_18.04.zip
